mypy/nodes.py

```diff
@@ -2403,6 +2400,140 @@ def __getattribute__(self, attr: str) -> None:
         raise AssertionError('De-serialization failure: TypeInfo not fixed')
 
 
+class TypeAlias(SymbolNode):
+    """
+    A sybmol node representing type alias.
+
+    Type alias is a static concept, in contrast to variables with variables
+    with types like Type[...]. Namely:
+        * type aliases
+            - can be used in type context (annotations)
+            - cannot be re-assigned
+        * variables with type Type[...]
+            - cannot be used in type context
+            - but can be re-assigned
+
+    An alias can be defined only by an assignment to a name (not any other lvalues).
+
+    Such assignment defines an alias by default. To define a variable,
+    an explicit Type[...] annotation is required. As an exception,
+    at non-global scope non-subscripted rvalue creates a variable even without
+    an annotation. This exception exists to accommodate the common use case of
+    class-valued attributes, see SemanticAnalyzer.check_and_set_up_type_alias
+    for details.
+
+    Aliases can be generic. Currently, mypy uses unbound type variables for
+    generic aliases and identifies them by name. Essentially, type aliases
+    work as macros that expand textually. The definition and expansion rules are
+    following:
+
+        1. An alias targeting a generic class without explicit variables act as
+        the given class:
+
+            A = List
+            AA = List[Any]
+
+            x: A  # same as List[Any]
+            x: A[int]  # same as List[int]
+
+            x: AA  # same as List[Any]
+            x: AA[int]  # Error!
+
+        2. An alias using explicit type variables in its rvalue expects
+        replacements (type arguments) for these variables. If missing, they
+        are treated as Any, like for other generics:
+
+            B = List[Tuple[T, T]]
+
+            x: B  # same as List[Tuple[Any, Any]]
+            x: B[int]  # same as List[Tuple[int, int]]
+
+            def f(x: B[T]) -> T: ...  # without T, Any would be used here
+
+        3. An alias can be defined using another aliases. In the definition
+        rvalue the Any substitution doesn't happen for top level unsubscripted
+        generics:
```

Comment:
It turns out these were prohibited, but I think we can allow and interpret as `Tuple[Any, ...]` and `Callable[..., Any]` (one line fix). Things like `U = Union` are still prohibited, but work at runtime. It will take some time to allow this and no one complained about this (also one can still write `from typing import Union as U`) so I would postpone this.